[FLINK-17599][docs] Update documents due to FLIP-84 (#12577)
apache/flink: docs/dev/table/common.md

a Table API or SQL query is translated when:

* `TableEnvironment.executeSql()` is called. This method is used for executing a given statement, and the sql query is translated immediately once this method is called.
* `Table.executeInsert()` is called. This method is used for inserting the table content to the given sink path, and the Table API is translated immediately once this method is called.
* `Table.execute()` is called. This method is used for collecting the table content to local client, and the Table API is translated immediately once this method is called.
* `SatementSet.execute()` is called. A `Table` (emitted to a sink through `SatementSet.addInsert()`) or an INSERT statement (specified through `SatementSet.addInsertSql()`) will be buffered in `SatementSet` first. They are translated once `SatementSet.execute()` is called. All sinks will be optimized into one DAG.
* A `Table` is translated when it is converted into a `DataStream` (see [Integration with DataStream and DataSet API](#integration-with-datastream-and-dataset-api)). Once translated, it's a regular DataStream program and is executed when `StreamExecutionEnvironment.execute()` is called.

> **Note:** Since 1.11 version, `sqlUpdate()` and `insertInto()` are deprecated. If the table program is built from these two methods, we must use `StreamTableEnvironment.execute()` method instead of `StreamExecutionEnvironment.execute()` method to execute it.

Reviewer: method `sqlUpdate()` and `insertInto()`

1. Optimization of the logical plan
2. Translation into a DataStream or DataSet program

A Table API or SQL query is translated when:

* `TableEnvironment.executeSql()` is called. This method is used for executing a given statement, and the sql query is translated immediately once this method is called.
* `Table.executeInsert()` is called. This method is used for inserting the table content to the given sink path, and the Table API is translated immediately once this method is called.
* `Table.execute()` is called. This method is used for collecting the table content to local client, and the Table API is translated immediately once this method is called.

> * `SatementSet.execute()` is called. A `Table` (emitted to a sink through `SatementSet.addInsert()`) or an INSERT statement (specified through `SatementSet.addInsertSql()`) will be buffered in `SatementSet` first. They are translated once `SatementSet.execute()` is called. Each sink will be optimized independently. The execution graph contains multiple independent sub-DAGs.

Reviewer 2: typo for `StatementSet` at multiple locations

</div>
</div>

**Note:** A detailed discussion about dynamic tables and their properties is given in the [Dynamic Tables](streaming/dynamic_tables.html) document. 

> <span class="label label-danger">Attention</span> **Once the Table is converted to a DataStream, we must use the StreamExecutionEnvironment.execute method to execute the DataStream program.**

Reviewer 3: ```suggestion
<span class="label label-danger">Attention</span> **Once the Table is converted to a DataStream, please use the `StreamExecutionEnvironment.execute` method to execute the DataStream program.**
```
